docgen/plugins/rev-assets.js

```diff
@@ -0,0 +1,69 @@
+import fs from 'fs';
+import path from 'path';
+import glob from 'glob';
+import hasha from 'hasha';
+import replace from 'replace-in-file';
+
+import config from '../config.js';
+
+const DIST_PATH = path.resolve(config.docsDist);
+
+// HELPERS TO ADD REVISION HASH TO FILENAMES
+// -----------------------------------------
+function computeHashForFiles(files: string[]): Promise<[{ file: string, hash: string }]> {
+  const promises = files.map(
+    file => hasha
+      .fromFile(file, {algorithm: 'md5'})
+      .then(hash => ({file, hash}))
+  );
+
+  return Promise.all(promises);
+}
+
+function renameFiles(files: { file: string, hash: string }[]): { oldPath: string, newPath: string }[] {
+  return files.map(({file, hash}) => {
+    const ext = path.extname(file);
+    const newPath = `${file.replace(ext, `-${hash}`)}${ext}`;
+
+    // rename original with hash file
+    fs.renameSync(file, newPath);
+
+    // check for `.[css|js].map` files to rename also
+    if ((ext === '.css' || ext === '.js') && fs.existsSync(`${file}.map`)) {
+      fs.renameSync(`${file}.map`, `${newPath}.map`);
+    }
+
+    return {oldPath: file, newPath};
+  });
+}
+
+function renameReferences(inFiles: string|string[], files: { oldPath: string, newPath: string }[]) {
+  const {from, to} = files.reduce((toRename, {oldPath, newPath}) => ({
+    from: [...toRename.from, new RegExp(oldPath.replace(`${DIST_PATH}/`, ''), 'g')],
+    to: [...toRename.to, newPath.replace(`${DIST_PATH}/`, '')],
+  }), {from: [], to: []});
+
+  return replace({from, to, files: inFiles});
+}
+
+export default function revAssets(): Promise<*> {
+  // COMPUTE CSS
+  // -----------
+  const CSS_PATH = path.join(DIST_PATH, 'stylesheets');
+  const CSS_FILES = glob.sync(`${CSS_PATH}/**/*.css`);
+
+  const computeCSSFiles = () => computeHashForFiles(CSS_FILES)
```

Comment:
I think if you want to be safe you should:
- compute MD5 for images
- replace in CSS, JS, HTML
- compute MD5 for CSS
- replace in JS, HTML
- compute MD5 FOR JS
- replace in HTML

This way you should be safe. Maybe from our setup this is already safe the way you did it?